server/rest/src/main/java/org/infinispan/rest/server/InfinispanResponse.java

```diff
@@ -0,0 +1,153 @@
+package org.infinispan.rest.server;
+
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+
+import org.infinispan.rest.server.operations.mediatypes.MediaType;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+import io.netty.buffer.Unpooled;
+import io.netty.handler.codec.http.DefaultFullHttpResponse;
+import io.netty.handler.codec.http.FullHttpResponse;
+import io.netty.handler.codec.http.HttpHeaderNames;
+import io.netty.handler.codec.http.HttpHeaderValues;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import io.netty.handler.codec.http.HttpUtil;
+import io.netty.handler.codec.http.HttpVersion;
+import io.netty.handler.codec.http2.HttpConversionUtil;
+
+public class InfinispanResponse {
```

Comment:
@slaskawi The question is, do we really need a wrapper for `NettyHttpResponse`? :)

In any case, I'd remove the optionals, because escape analysis is not reliable enough. OTOH it requires the lambdas to be inlined (objects passed as arguments to other methods are allocated on heap), but then it's automatically disabled if the method gets too big because of inlining: http://vanillajava.blogspot.ro/2015/01/java-lambdas-and-low-latency.html. So you might get 0 allocations when you test in JMH, but then lots of allocations if `toNettyHttpResponse` gets inlined into a big method.